### PR TITLE
Reporting Test Serverless structure: api_integration

### PR DESCRIPTION
**Pull request by tsullivan**
## Summary

This PR adds some structure for API Integration tests for Reporting in serverless. I've added a single test to get started, a test to ensure users can only delete reports they've created.

Initially I planned to add a functional test for report deletion, but currently this is blocked: if we want to use our existing archive of reports, we need to log in as `test_user` which is the user that created the archived reports. In serverless functional tests, it looks like we're currently only able to test with the `elastic` user and that doesn't help us.

Next step: The next API Integration tests that will be ready to work on can be copied from:
* x-pack/test/reporting_api_integration/reporting_and_security/csv_v2.ts
* x-pack/test/reporting_api_integration/reporting_and_security/error_codes.ts
* x-pack/test/reporting_api_integration/reporting_without_security/job_apis_csv.ts
* x-pack/test/reporting_api_integration/reporting_and_security/generate_csv_discover.ts
* x-pack/test/reporting_api_integration/reporting_and_security/security_roles_privileges.ts

Note: I would like to hold off on adding tests and services for `usage`, we might not need it.

### CODE REVIEW

File: x-pack/test_serverless/tsconfig.json [hunk 1]
```diff
@@ -46,5 +46,8 @@
     "@kbn/test-subj-selector",
     "@kbn/reporting-plugin",
     "@kbn/rison",
+    "@kbn/core-http-common",
+    "@kbn/data-views-plugin",
+    "@kbn/core-saved-objects-server",
```
Comment (tsullivan) [inline x-pack/test_serverless/tsconfig.json:51]: not sure how these lines were added 🤨

File: x-pack/test_serverless/functional/test_suites/common/reporting/management.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import { FtrProviderContext } from '../../../ftr_provider_context';
+
+export default ({ getPageObjects, getService }: FtrProviderContext) => {
+  // const esArchiver = getService('esArchiver');
+  // const log = getService('log');
+  // const pageObjects = getPageObjects(['common', 'security']);
+  // const retry = getService('retry');
+  // const testSubjects = getService('testSubjects');
+
+  describe('Reporting Management', function () {});
```
Comment (tsullivan) [inline x-pack/test_serverless/functional/test_suites/common/reporting/management.ts:17]: There isn't much we can do in functional tests until we have the ability to log in as a non-`elastic` user.

File: .buildkite/ftr_configs.yml [hunk 3]
```diff
@@ -87,12 +87,10 @@ disabled:
   - x-pack/test_serverless/api_integration/test_suites/observability/config.ts
   - x-pack/test_serverless/api_integration/test_suites/search/config.ts
   - x-pack/test_serverless/api_integration/test_suites/security/config.ts
-  - x-pack/test_serverless/api_integration/test_suites/reporting/config.ts
   - x-pack/test_serverless/functional/test_suites/common/config.ts
   - x-pack/test_serverless/functional/test_suites/observability/config.ts
   - x-pack/test_serverless/functional/test_suites/search/config.ts
   - x-pack/test_serverless/functional/test_suites/security/config.ts
-  - x-pack/test_serverless/functional/test_suites/reporting/config.ts
```
Comment (tsullivan) [inline .buildkite/ftr_configs.yml:95]: By design, the only serverless test suites are: `observability`, `search`, `security` and `common`. Reporting tests should go under `common`.

File: x-pack/test_serverless/api_integration/services/reporting_api.ts [hunk 4]
```diff
@@ -1,22 +0,0 @@
...
-export function ReportingApiProvider(context: FtrProviderContext) {
-  return {
-    ...createScenarios(context),
-    ...createUsageServices(context),
-  };
-}
-
-export const services = {
-  supertestWithoutAuth: apiIntegrationServices.supertestWithoutAuth,
-  usageAPI: apiIntegrationServices.usageAPI,
```
Comment (tsullivan) [inline x-pack/test_serverless/api_integration/services/reporting_api.ts:21]: I would like to hold off on adding `usage` tests and services for now.

**Review by rshen91 — APPROVED** [review]

**Pull request merged**